windows-installer/openssl.nsi

```diff
@@ -1,111 +1,144 @@
+# NSIS windows installer script file
+#
+# Requirements: NSIS 3.0 must be installed with the MUI plugin
+# Usage notes:
+# This script expects to be executed from the directory it is
+# currently stored in.  It expects a 32 bit and 64 bit windows openssl
+# build to be present in the ..\${BUILD32} and ..\${BUILD64} directories
+# respectively
+# ####################################################
+
+!include "MUI.nsh"
+!include "winmessages.nsh"
+
+!define PRODUCT_NAME "OpenSSL"
+!define VERSION "${MAJOR}.${MINOR}.${PATCH}"
+
+# The name of the output file we create when building this
+# NOTE version is passed with the /D option on the command line
+OutFile "openssl-${VERSION}-installer.exe"
+
+# The name that will appear in the installer title bar
+NAME "${PRODUCT_NAME} ${VERSION}"
+
+ShowInstDetails show
+
+
+Var DataDir
+Var ModDir
+
+Function .onInit
+    StrCpy $INSTDIR "C:\Program Files\openssl-${MAJOR}.${MINOR}"
+FunctionEnd
+
+# This section is run if installation of 32 bit binaries are selected
+
+!ifdef BUILD64
+# This section is run if installation of the 64 bit binaries are selectd
+SectionGroup "64 Bit Installation"
+    Section "64 Bit Binaries"
+        SetOutPath $INSTDIR\x64\lib
+        File /r "${BUILD64}\Program Files\OpenSSL\lib\"
+        SetOutPath $INSTDIR\x64\bin
+        File /r "${BUILD64}\Program Files\OpenSSL\bin\"
+        SetOutPath "$INSTDIR\x64\Common Files"
+        File /r "${BUILD64}\Program Files\Common Files\"
+    SectionEnd
+    Section "x64 Development Headers"
+        SetOutPath $INSTDIR\x64\include
+        File /r "${BUILD64}\Program Files\OpenSSL\include\"
+    SectionEnd
+SectionGroupEnd
+!endif
+
+!ifdef BUILD32
+# This section is run if installation of the 64 bit binaries are selectd
+SectionGroup "32 Bit Installation"
+    Section "32 Bit Binaries"
+        SetOutPath $INSTDIR\x32\lib
+        File /r "${BUILD32}\Program Files (x86)\OpenSSL\lib\"
+        SetOutPath $INSTDIR\x32\bin
+        File /r "${BUILD32}\Program Files (x86)\OpenSSL\bin\"
+        SetOutPath "$INSTDIR\x64\Common Files"
+        File /r "${BUILD32}\Program Files (x86)\Common Files\"
+    SectionEnd
+    Section "x32 Development Headers"
+        SetOutPath $INSTDIR\x32\include
+        File /r "${BUILD32}\Program Files (x86)\OpenSSL\include\"
+    SectionEnd
+SectionGroupEnd
+!endif
+
+!ifdef BUILD64
+Section "Documentation"
+    SetOutPath $INSTDIR\html
+    File /r "${BUILD64}\Program Files\OpenSSL\html\"
+SectionEnd
+!endif
+
+# Always install the uninstaller and set a registry key
+Section
+    WriteUninstaller $INSTDIR\uninstall.exe
+SectionEnd
+
+!define env_hklm 'HKLM "SYSTEM\CurrentControlSet\Control\Session Manager\Environment"'
+!define openssl_hklm 'HKLM "SOFTWARE\OpenSSL-${MAJOR}.${MINOR}-${CTX}"'
+
+# This is run on uninstall
+Section "Uninstall"
+    RMDIR /r $INSTDIR
+    DeleteRegValue ${openssl_hklm} OPENSSLDIR
+    DeleteRegValue ${openssl_hklm} MODULESDIR
+    DeleteRegValue ${openssl_hklm} ENGINESDIR
```

Comment:
I see these reg keys get uninstalled here - but somehow I'm missing the location where they get inserted in the first place?

Reply:
Its in CheckRunUninstaller